packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx

```diff
@@ -127,11 +127,17 @@ describe('NewUserEmailForm test', () => {
       subject = render(
         <WrapNewUserEmailForm accountExistsReturnValue={false} />
       );
+
       const firstEmail = subject.getByTestId('new-user-email');
       const secondEmail = subject.getByTestId('new-user-confirm-email');
       fireEvent.change(firstEmail, { target: { value: 'valid@email.com' } });
-      fireEvent.change(secondEmail, {
-        target: { value: 'not.the.same@email.com' },
+
+      // need to wait for the first event on secondEmail before calling blur
+      // otherwise blur will discard the change event
+      await act(async () => {
+        fireEvent.change(secondEmail, {
+          target: { value: 'not.the.same@email.com' },
+        });
```

Comment:
I'm not sure why your `await act...` works (I'm guessing it just slows down the script enough for the async validation on the first input field to complete before firing the next event), but I think the actual root cause is that we're not waiting for the async validation from the first input field's change event to complete.

I tried this diff locally in your feature branch, and it passes -- perhaps this would be a more clear fix for this test failure? This would match similar test patterns we have where we mock (async) API calls.

```diff
diff --git a/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx b/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
index 541e3f147..6a9d0e8cf 100644
--- a/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
+++ b/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
@@ -12,8 +12,13 @@ jest.mock('../../lib/apiClient', () => ({
 }));
 import { apiFetchAccountStatus } from '../../lib/apiClient';
 
-import { NewUserEmailForm, emailInputValidationAndAccountCheck } from './index';
+import {
+  NewUserEmailForm,
+  emailInputValidationAndAccountCheck,
+  checkAccountExists,
+} from './index';
 import { Localized } from '@fluent/react';
+import waitForExpect from 'wait-for-expect';
 const selectedPlan = {
   plan_id: 'planId',
   plan_name: 'Pro level',
@@ -53,10 +58,8 @@ const WrapNewUserEmailForm = ({
         setEmailsMatch={setEmailsMatch}
         setValidEmail={setValidEmail}
         setAccountExists={setAccountExists}
+        checkAccountExists={checkAccountExists}
         getString={(id: string) => id}
-        checkAccountExists={() =>
-          Promise.resolve({ exists: accountExistsReturnValue })
-        }
         selectedPlan={selectedPlan}
       />
     </div>
@@ -131,13 +134,11 @@ describe('NewUserEmailForm test', () => {
       const firstEmail = subject.getByTestId('new-user-email');
       const secondEmail = subject.getByTestId('new-user-confirm-email');
       fireEvent.change(firstEmail, { target: { value: 'valid@email.com' } });
-
-      // need to wait for the first event on secondEmail before calling blur
-      // otherwise blur will discard the change event
-      await act(async () => {
-        fireEvent.change(secondEmail, {
-          target: { value: 'not.the.same@email.com' },
-        });
+      await waitForExpect(() =>
+        expect(apiFetchAccountStatus).toHaveBeenCalledTimes(1)
+      );
+      fireEvent.change(secondEmail, {
+        target: { value: 'not.the.same@email.com' },
       });
       fireEvent.blur(secondEmail);
     });
```

Reply:
per the discussion in slack, we went with removing the wrapping act call on the body of the test - we think that might have been the main issue as the test is pretty straight forward and works as expected without that superfluous act.

let me know if this is satisfactory from your point of view.

Reply:
Hmm true that `render` is also already wrapped in `act` and is sync, just like `fireEvent`. If that works then that's great. I guess we don't actually need to wait for the async validation on the first email input here, since our second email input's validation is based on the value of the first input field, not the result of its async validation.

WFM thanks.